tests/core/notification/test_notifier.py

```diff
@@ -250,9 +250,10 @@ def test_publish_event():
 
     scenario = tp.create_scenario(scenario_config)
     cycle = scenario.cycle
-    pipeline = scenario.pipelines["pipeline_config"]
     task = scenario.tasks[task_config.id]
     dn = scenario.data_nodes[dn_config.id]
+    # NOTE: pipeline is created only when accessed via scenario.pipelines
```

Comment:
The note seems not True anymore. No? We are checking the pipeline creation event later.

Reply:
it's still True actually. When we call the getter `scenario.pipelines`, it will trigger the `__get_pipelines` function which then create all pipelines of the scenario and trigger the CREATE event

Reply:
AH ok!
Got it